libMachO/Core/core.h

```diff
@@ -38,10 +38,36 @@
 #include "base.h"
 
 #include <mach/mach.h>
-#if TARGET_OS_MAC && !TARGET_OS_IPHONE
-#   include <mach/mach_vm.h>
-#   define MK_HAVE_MACH_VM 1
-#endif
+
+kern_return_t mach_vm_allocate
```

Comment:
I am against using the `mach_*` functions on embedded because they are not included in the public SDK headers.  Mach-O Kit can switch to using the `vm_` APIs if they provide the required functionality.  Or gate usage of the `mach_*` APIs behind a compile flag like I suggested in my original comment.

Reply:
Would you be fine with switching to `vm_*` APIs on macOS as well? It might be messy to use both APIs internally since there would need to be preprocessor checks to determine which API to use, and additionally the types are slightly different. The outward facing types will still be `mach_` based as the types themselves are defined even on iOS.

Reply:
FWIW this change turned out to be easier than expected and I've implemented it in 430a952. If you'd rather use `mach_vm_*` on macOS, I can add that back with a preprocessor check for the macOS platform.